docs/sections/user_guide/yaml/components/filter_topo.rst

```diff
@@ -20,10 +20,18 @@ config:
 
 Configuration parameters for the ``orog_gsl`` component.
 
+  **filtered_orog:**
+
+  Name of the filtered output file.
+
```

Comment:
I added this as a configurable item, although its name is decided in the filter_topo program. We need access to the name downstream, so I thought it might as well be defined here.

Reply:
Am I right in thinking that this is just created in the run directory, such that this really is just a filename and not a path?

Also, I think this came up in another context, but I suspect it's a bad idea to expose in configuration files items that are not configurable, if that's what you mean. If we _had_ to do this, then at least I suppose the schema should enforce exactly the one correct value. But I feel like this is component-specific knowledge that should be hidden away in its driver. Could it be a class attribute or static method on the driver class in question, so that other drivers could consult it if they need to know it?

Reply:
Yeah, I don't know what to do about this one. The exact name needs to be created based on formulaic knowledge that the experiment YAML has access to (tile number, CRES, etc.). We could duplicate that piece of information from the Fortran code inside the driver instead of putting it here. The Fortran code that does it looks like:

```
tile_file = trim(topo_file)//'.tile'//trim(text)//'.nc'
```

Where `text` is the tile number, and `topo_file` is the file path prefix specified in the namelist. When we expand to include support for a global domain, we'll need this to include all 6 files (6 tiles) that are copied in. 

I'm stuck on this one...I can def see the perks of both sides. Currently, the driver objects are not communicating their output file name(s), but it is certainly a nice to have feature. For all drivers, we currently have to set it in the config as _a priori_ knowledge. Maybe could leave it that way in this iteration and return to this as a moderate upgrade for all drivers?

Reply:
Since it has to be constructed based on the config, it'd have to be provided via an instance method instead of a class/static method. But I'm still in favor of doing this -- constructing the filename as the Fortran code you show does -- in the driver, and would vote for not doing a release where users have to construct this ultimately derivative value. If we don't do it now, could we plan on fixing it before a v2.5.0 release?

Reply:
I guess it could still be a static/class method, with the components of the filename passed in as (maybe some optional) arguments. If an instance of the driver called it, it could pass in the values from its config, but non-driver-instance callers could pass in the filename components. I think that's still better than having users just provide the filename, as it makes it clear that the filename is not arbitrary, makes sure it's of the right form, and makes the connection between the filename and the overall configuration obvious.